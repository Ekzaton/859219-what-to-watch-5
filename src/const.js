export const APIRoute = {
  FILMS: `/films/`,
  FAVORITE: `/favorite/`,
  PROMO: `/films/promo/`,
  COMMENTS: `/comments/`,
  LOGIN: `/login/`
};

export const AppRoute = {
  ROOT: `/`,
  SIGN_IN: `/login/`,
  MY_LIST: `/mylist/`,
  REVIEW: `/review/`,
  REVIEW_ID: `/films/:id/review/`,
  PLAYER: `/player/`,
  PLAYER_ID: `/player/:id/`,
  FILMS: `/films/`,
  FILMS_ID: `/films/:id/`
};

export const Tab = {
  OVERVIEW: `overview`,
  DETAILS: `details`,
  REVIEWS: `reviews`
};

export const RatingLevel = {
  BAD: `Bad`,
  NORMAL: `Normal`,
  GOOD: `Good`,
  VERY_GOOD: `Very good`,
  AWESOME: `Awesome`,
  UNRATED: `Unrated`
};

export const Review = {
  MIN_LENGTH: 50,
  MAX_LENGTH: 400,
};

export const AuthorizationStatus = {
  AUTH: `AUTH`,
  NO_AUTH: `NO_AUTH`,
};

export const DEFAULT_GENRE = `All genres`;

export const DEFAULT_MOVIES_COUNT = 8;
