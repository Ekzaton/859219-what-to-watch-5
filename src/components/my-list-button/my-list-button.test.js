import React from "react";
import renderer from "react-test-renderer";
import {Provider} from "react-redux";
import configureStore from "redux-mock-store";

import {AuthorizationStatus} from "../../const";

import MyListButton from "./my-list-button";

describe(`MyListButton snapshot testing`, () => {
  it(`(without Auth) renders store-connected component correctly`, () => {
    const mockStore = configureStore([]);

    const store = mockStore({
      APP_USER: {
        status: AuthorizationStatus.NO_AUTH,
      },
    });

    const myListButtonComponent = renderer.create(
        <Provider store={store}>
          <MyListButton
            id={5}
            isFavorite={true}
          />
        </Provider>
    );

    expect(myListButtonComponent.toJSON()).toMatchSnapshot();
  });

  it(`(with Auth) renders store-connected component correctly`, () => {
    const mockStore = configureStore([]);

    const store = mockStore({
      APP_USER: {
        status: AuthorizationStatus.AUTH,
      },
    });

    const myListButtonComponent = renderer.create(
        <Provider store={store}>
          <MyListButton
            id={5}
            isFavorite={true}
          />
        </Provider>
    );

    expect(myListButtonComponent.toJSON()).toMatchSnapshot();
  });
});
