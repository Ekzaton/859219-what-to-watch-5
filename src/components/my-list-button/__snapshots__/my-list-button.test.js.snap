// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyListButton snapshot testing (with Auth) renders store-connected component correctly 1`] = `
<button
  className="btn btn--list movie-card__button"
  onClick={[Function]}
  type="button"
>
  <svg
    height="14"
    viewBox="0 0 18 14"
    width="18"
  >
    <use
      xlinkHref="#in-list"
    />
  </svg>
  <span>
    My list
  </span>
</button>
`;

exports[`MyListButton snapshot testing (without Auth) renders store-connected component correctly 1`] = `null`;
